cmd/ctr/commands/tasks/exec.go

```diff
@@ -86,11 +91,22 @@ var execCommand = cli.Command{
 		pspec.Terminal = tty
 		pspec.Args = args
 
-		cioOpts := []cio.Opt{cio.WithStdio, cio.WithFIFODir(context.String("fifo-dir"))}
-		if tty {
-			cioOpts = append(cioOpts, cio.WithTerminal)
+		var ioCreator cio.Creator
+
+		if logURI := context.String("log-uri"); logURI != "" {
+			uri, err := url.Parse(logURI)
+			if err != nil {
+				return err
+			}
+			ioCreator = cio.LogURI(uri)
+		} else {
+			cioOpts := []cio.Opt{cio.WithStdio, cio.WithFIFODir(context.String("fifo-dir"))}
```

Comment:
should we need to return error if both `fifo-dir` and `log-uri` are set?

Reply:
Done